ouroboros-consensus/docs/report/chapters/future/genesis.tex

```diff
@@ -1337,6 +1337,141 @@ \subsection{Block production}
 not up to date may in fact be helpful for an adversary. We should therefore
 disable block production while the node is not up to date.
 
+\section{Dynamic genesis window length}
+
+The genesis window length is defined to be equal to $3k/f$. However this
+definition only makes sense in Praos and beyond as such protocols actually have
+a $f$ active slot coefficient. There is no equivalent $f$ for (P)BFT.
+
+It is important to consider several things here:
+
+\begin{itemize}
+\item The $f = 0.05$ factor is actually present in (P)BFT already as slots are 20
+  seconds long.
+\item The genesis window length is bounded from the top by the forecast
+  range. It wouldn't be practical to define a genesis window length that
+  involved blocks that we possibly could not verify. Currently the forecast
+  range is $3k/f$ slots in Praos and $2k$ slots in (P)BFT.
+\item The genesis window length is bounded from the bottom by the honest chain
+  growth rate. Currently we require that the window always has at least $k$
+  blocks inside the window and this is achieved in Praos in $3k/f$ slots and in
+  $2k$ slots in (P)BFT.
+\end{itemize}
+
+Because of the reasons above, we could have the following bounds:
+\begin{itemize}
+\item In PBFT: $2k \le s \le 2k$
+\item In Praos: $3k/f \le s \le 3k/f$
+\end{itemize}
+
+\subsection{The current decision}
+
+However one can argue that in reality and with the use cases that are envisioned
+for this consensus layer, it doesn't currently provide much benefits
+implementing the machinery to be able to achieve this change in the genesis
+window length.
+
+In (P)BFT, the chain was produced by a set of core nodes that were extending the
+chain in a round-robin like fashion, and when one is to bootstrap from genesis,
+only one chain will be served as only one existed. There will be no choice to
+make and therefore the genesis window will never ever be invoked. Therefore
+whatever value we use for (P)BFT is irrelevant.
+
+Furthermore, if someone uses this consensus layer to run something different
+that the actual Cardano network, either it doesn't make sense to use (P)BFT
+because they want a decentralized network (so $s$ can be set to $3k/f$) or they
+are only going to run (P)BFT and there block production is controlled by the
+core nodes and there will be no choices to make.
+
+We then decided to use $3k/f$ with the values those variables have in Praos to
+define the genesis window length as up to the current protocol choices ((P)BFT -
+TPraos - Praos). The genesis window length will then be just a parameter in a
+way similar to $k$ and will be fixed during the execution.
+
+\subsection{Future approaches}
+
+There is however a subtlety that has to be made precise here. This choice is
+correct for now and as long as $f$ doesn't change. Currently, the active slots
+coefficient is not a votable parameter so there is no current way to change it.
+However, if it was made votable or even era-dependent, further development will
```

Comment:
Further investigation?

Reply:
I see that you provided a possible implementation, but are we sure this is sound?